libstorage/CachedStorage.cpp

```diff
@@ -571,7 +550,38 @@ void CachedStorage::init()
         m_ID = boost::lexical_cast<size_t>(numStr);
     }
 
-    startClearThread();
+    if (!disabled())
+    {
+        startClearThread();
+    }
+}
+
+void CachedStorage::stop()
+{
+    STORAGE_LOG(INFO) << "Stoping flushStorage thread";
+    m_running->store(false);
+
+    m_taskThreadPool->stop();
+
+    if (m_clearThread)
+    {
+        if (m_clearThread->get_id() != std::this_thread::get_id())
```

Comment:
stop() called by m_clearThread should be impossible. I think this modify is unnecessary😢